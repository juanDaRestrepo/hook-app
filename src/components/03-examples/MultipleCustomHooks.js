import React from 'react'
import { useCounter } from '../../Hooks/useCounter';
import { useFetch } from '../../Hooks/useFetch';
import '../02-useEffect/effects.css';
export const MultipleCustomHooks = () => {

    const {counter,increment,decrement} = useCounter(1);
    
   const {loading,data}=useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
   
   const {author,quote} = !!data && data[0];
   
   return (
        <div>
            <h1>BreakingBad Quotes</h1>
            <hr />
            {
                loading?
                (
                    <div className='alert alert-info text-center'>
                        loading...
                    </div>
                )
                :
                (
                    <blockquote className='blockquote text-end'>
                    <p className='mb-1'>{quote}</p>
                    <footer className='blockquote-footer'>{author}</footer>
                    </blockquote>
                )
            }
            
            <button className='btn btn-primary' onClick={increment}>
                Siguiente quote
            </button>

            <button className='btn btn-primary ms-2' onClick={decrement}>
                Anterior quote
            </button>
            
        </div>
    )
}
