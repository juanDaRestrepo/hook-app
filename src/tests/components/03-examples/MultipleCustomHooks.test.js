import React from 'react';
import { shallow } from 'enzyme';
import {MultipleCustomHooks} from "../../../components/03-examples/MultipleCustomHooks";
import setupTest from '../../../setupTest';
import { useFetch } from '../../../Hooks/useFetch';

describe('Pruebas en <MultipleCustomHooks />' , () => {

    test('Debe de mostrarse correctamente', () => {

        const wrapper = shallow(<MultipleCustomHooks />);
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe mostar la informaciÃ³n', () => {
      
    });
    
})