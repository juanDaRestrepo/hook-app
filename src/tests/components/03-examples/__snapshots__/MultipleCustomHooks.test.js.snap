// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary "
    onClick={[Function]}
  >
    Anterior quote
  </button>
  <button
    className="btn btn-primary ms-2"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
